textract2page/convert_aws.py

```diff
@@ -720,16 +785,99 @@ def convert_file(
                         )
                     ),
                     id=f"word-{word.id}",
+                    production=word.text_type,
                 )
                 if word.text:
-                    pagexml_word.add_TextEquiv(TextEquivType(
-                        conf=word.confidence,
-                        Unicode=word.text))
+                    pagexml_word.add_TextEquiv(
+                        TextEquivType(conf=word.confidence, Unicode=word.text)
+                    )
+                pagexml_text_line.add_Word(pagexml_word)
+
+    for layout in layouts:
+        # ignore layout_type: other
+        if layout.textract_layout_type == "LAYOUT_FIGURE":
+            pagexml_text_region = ImageRegionType(
+                Coords=CoordsType(
+                    points=points_from_aws_geometry(
+                        layout.geometry, pil_img.width, pil_img.height
+                    )
+                ),
+                id=f"layout-image-region-{layout.id}",
+                type_=layout.page_layout_type,
+                custom=f"textract-layout-type: {layout.textract_layout_type.split('LAYOUT_')[1].lower()};",
+            )
+            pagexml_page.add_TextRegion(pagexml_text_region)
+
+            if preserve_reading_order:
+                global_ordered_group.add_RegionRefIndexed(
+                    RegionRefIndexedType(
+                        index=global_reading_order_index,
+                        regionRef=f"layout-text-region-{layout.id}",
+                    )
+                )
+                global_reading_order_index += 1
+            continue
+        if layout.textract_layout_type == "LAYOUT_TABLE":
+            # we cover tables separatly
+            continue
```

Comment:
are you sure about that already?

So if AWS comes back with a PAGE → LAYOUT_TABLE → TABLE → ... block structure, do we get a TableRegion, or will it be lost (because the top link is missing)?

Reply:
pretty much, we should test it nevertheless.